npm/vite-dev-server/src/plugins/cypress.ts

```diff
@@ -52,21 +53,45 @@ export const Cypress = (
     configResolved (config) {
       base = config.base
     },
-    async transformIndexHtml () {
+    async transformIndexHtml (html) {
+      // it's possibe other plugins have modified the HTML
+      // before we get to. For example vitejs/plugin-react will
+      // add a preamble. We do our best to look at the HTML we
+      // receive and inject it.
+      // For now we just grab any `<script>` tags and inject them to <head>.
+      // We will add more handling as we learn the use cases.
+      const root = parse(html)
+
+      const scriptTagsToInject = root.childNodes.filter((node) => {
+        // @ts-ignore
+        return node.rawTagName === 'script'
+      })
+
       const indexHtmlPath = resolve(projectRoot, indexHtmlFile)
 
       debug('resolved the indexHtmlPath as', indexHtmlPath, 'from', indexHtmlFile)
-      const indexHtmlContent = await fs.promises.readFile(indexHtmlPath, { encoding: 'utf8' })
+
+      let indexHtmlContent = await fs.promises.readFile(indexHtmlPath, { encoding: 'utf8' })
+
+      // Inject the script tags
+      indexHtmlContent = indexHtmlContent.replace(
+        '<head>',
+        `<head>
+        ${scriptTagsToInject.map((script) => script.toString())}
+      `,
+      )
```

Comment:
Both fine, I think I like `let` + reassign a little more than `then`, but good to know this is an option, too.

Thoughts on mixing `await` and `then`? I usually pick one or the other, but maybe combining them can make code more readable (and avoid mutation).

Reply:
I agree that it's usually best to avoid mixing `async/await` and `then` paradigms. I guess there is one other option to just wrap the await statement with parens but that isn't super-elegant either, so feel free to keep it the way it is:

```
const indexHtmlContent = (await fs.promises.readFile(indexHtmlPath, { encoding: 'utf8' }))
        // Inject the script tags
        .replace(
          '<head>',
          `<head>
          ${scriptTagsToInject.map((script) => script.toString())}
        `,
        )
```